src/core/thread/mle.cpp

```diff
@@ -1783,7 +1783,7 @@ ThreadError Mle::HandleChildIdResponse(const Message &aMessage, const Ip6::Messa
 
         if ((mDeviceMode & ModeTlv::kModeFFD) && (numRouters < mMleRouter.GetRouterUpgradeThreshold()))
```

Comment:
Currently, either approach will work since `ProcessRouteTlv` will set the internal state appropriately for `GetActiveRouterCount`.  I think it is safer to compute the information directly from the Route TLV here given that `ProcessRouteTlv` will be stubbed out in a end-device-only optimized implementation.